eta/core/storage.py

```diff
@@ -805,13 +805,21 @@ def download_files_in_folder(
             filename = f["name"]
             file_id = f["id"]
             try:
-                local_path = os.path.join(local_dir, filename)
-                with etau.Timer() as t:
-                    self.download(file_id, local_path)
-                    filenames.append(filename)
-                logger.info(
-                    "File '%s' downloaded to '%s' (%s) (%d/%d)", filename,
-                    local_path, t.elapsed_time_str, idx, num_files)
+                if recursive and self.is_folder(file_id):
```

Comment:
calling `is_folder` here will be quite slow because you are calling it for every file, and Google Drive API requests are slow. Instead, you should be able to modify `list_files_in_folder` to include whether the object is a file or folder. Then you wouldn't need all these extra requests here.